File: 920-uncommon-words-from-two-sentences/uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> words = new HashMap<>();
        String[] str1 = s1.split(" ");
        String[] str2 = s2.split(" ");
        
        for (String str : str1) { 
            words.put(str, words.getOrDefault(str, 0) + 1);
        }
        for (String str : str2) {
            words.put(str, words.getOrDefault(str, 0) + 1);
        }

        List<String> uncommon = new ArrayList<>();
        for (Map.Entry<String, Integer> map : words.entrySet()) {
            if (map.getValue() == 1) {
                uncommon.add(map.getKey());
            }
        }

        return uncommon.toArray(new String[0]);
    }
}

File: 920-uncommon-words-from-two-sentences/uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        words = defaultdict(lambda: 0)
        for word in list(s1.split()):
            words[word] += 1
        for word in list(s2.split()):
            words[word] += 1

        result = []
        for key, value in words.items():
            if value == 1:
                result.append(key)

        return result